packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -148,6 +155,20 @@ export class SharedTreeCore<
 
     public submitEdit(edit: TChange): void {
         const delta = this.editManager.addLocalChange(edit);
+        // Edits performed before the first attach are treated as sequenced because they will be included
+        // in the attach summary that is uploaded to the service.
+        // Until this attach workflow happens, this instance essentially behaves as a centralized data structure.
+        if (this.detachedRevision !== undefined) {
+            const newRevision: SeqNumber = brand((this.detachedRevision as number) + 1);
+            const commit: Commit<TChange> = {
+                changeset: edit,
+                refNumber: this.detachedRevision,
+                seqNumber: newRevision,
```

Comment:
Maybe we should rename this field, since it doesn't always represent the sequence number anymore. I think we could also change its type to RevisionTag while we're at it.

This PR seems like an additional motivation to eventually switch to generating UUIDs for changesets instead of using sequence numbers to identify them.

Reply:
Agreed. Will make a separate PR for that.